test/gtest/ucs/test_datatype.cc

```diff
@@ -1356,3 +1359,215 @@ UCS_MT_TEST_F(test_datatype_ptr_map_safe, safe_put, num_threads)
         }
     }
 }
+
+
+class test_piecewise_func : public test_datatype {
+protected:
+    /* Represents piecewise function segment which consists of the linear
+     * function and the range on which this function is applicable.
+     * Both `start` and `end` is included to the range.
+     */
+    struct segment {
+        size_t            start;
+        size_t            end;
+        ucs_linear_func_t func;
+    };
```

Comment:
can we avoid it? seems like duplication of C structure

Reply:
Yes, we can. But it is not a duplication. I think it is more convenient to test ranges in such form, since:
1. add_range() func requires both start and end of the range to add.
2. there is no need to store info about previous segment to know the start of the segment.